### PR TITLE
Added promoteTransaction function to api.js

### PR DESCRIPTION
**Pull request by paulhandy**

### CODE REVIEW

File: lib/api/api.js [hunk 1]
```diff
@@ -606,6 +606,38 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
...
+*
+    * @returns the output of the final sendTransfer call
+*/
+    /* ----------------------------------------------------------------------------*/
+api.prototype.promoteTransaction = function(tail, depth, minWeightMagnitude, transfer, delay, maximum, callback) {
+    var self = this;
+    if (!inputValidator.isHash(tail)) {
+        return callback(errors.invalidTrytes());
+    }
+    self.sendTransfer(transfer[0].address, depth, minWeightMagnitude, transfers, {reference: tail}, function(err, res) {
```
Comment (alon-e) [inline lib/api/api.js]: `transfers` => `transfer`

File: lib/api/api.js [hunk 2]
```diff
@@ -606,6 +606,38 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
...
+        if (err == null && delay > 0 && maximum > 0) {
```
Comment (alon-e) [inline lib/api/api.js]: maximum is off by one.
for example: if you call w/ `maximum=1`, `sendTransfer()` would be called once, then enter the recursion, called again & finally stopped.

**Comment by paulhandy** [comment]
@th0br0 
> there's no way to cancel this
No, but it will only be performed `maximum` times
> does that reference parameter really work?
Now it does :trollface: 
> this keeps promoting even after confirmation
It does not check for confirmation - I would expect the wallet implementer to be responsible for confirmation status.

**Comment by alon-e** [comment]
>It does not check for confirmation - I would expect the wallet implementer to be responsible for confirmation status.
>there's no way to cancel this - No, but it will only be performed maximum times
but the wallet implementer cant cancel the promotion, only guess how many `maximum` times are needed till the next confirmation status test - or just do always call with `maximum = 1`.

**Comment by th0br0** [comment]
As this won't run just in node.js or browser processes but also mobile, please add a `isPromotable` method that essentially wraps `isTailConsistent` (API call to be impl by alon) and don't do any background work yourself?

File: lib/api/api.js [hunk 3]
```diff
@@ -606,6 +614,40 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
...
+    /* ----------------------------------------------------------------------------*/
+api.prototype.promoteTransaction = function(tail, depth, minWeightMagnitude, transfer, delay, maximum, callback) {
+    var self = this;
+    if (!inputValidator.isHash(tail)) {
+        return callback(errors.invalidTrytes());
+    }
+    if (maximum <= 0) goto promote_end;
+    self.sendTransfer(transfer[0].address, depth, minWeightMagnitude, transfer, {reference: tail}, function(err, res) {
+        if (err == null && delay > 0) {
+            setTimeout (delay, function() {
```
[the remark this answers is not in the document]
Comment (paulhandy) [reply lib/api/api.js]: good point

**Comment by paulhandy** [comment]
Do you prefer the wallet implementer to cancel or continue?

**Pull request merged**